ksql-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -57,23 +67,27 @@ private EngineExecutor(
 
   static EngineExecutor create(
       final EngineContext engineContext,
+      final ServiceContext serviceContext,
       final KsqlConfig ksqlConfig,
       final Map<String, Object> overriddenProperties
   ) {
-    return new EngineExecutor(engineContext, ksqlConfig, overriddenProperties);
+    return new EngineExecutor(engineContext, serviceContext, ksqlConfig, overriddenProperties);
   }
 
   ExecuteResult execute(final ConfiguredStatement<?> statement) {
     try {
       throwOnNonExecutableStatement(statement);
 
-      final QueryEngine queryEngine = engineContext.createQueryEngine();
+      final QueryEngine queryEngine = engineContext.createQueryEngine(serviceContext);
 
       final LogicalPlanNode logicalPlan = queryEngine.buildLogicalPlan(
           engineContext.getMetaStore(),
           statement.withConfig(ksqlConfig.cloneWithPropertyOverwrite(overriddenProperties))
       );
 
+      // DDL statements permissions are always denied at this point. Perhaps because the use of
+      // the SandboxedServiceContext.
```

Comment:
this seems like a bug - we should fix it by forwarding the correct access control methods to the true service context. You can see the proxies and what methods are being proxied to debug this.

Reply:
Fixed. I found the issue was that the `SandboxedKafkaTopicClient` was not returning a list of authorized operations. I created a `SandboxedTopicDescription` with pre-defined READ and WRITE operations if a `createTopic` is executed.